nordpool/elspot.py

```diff
@@ -27,88 +57,111 @@ def _parse_json(self, data, areas=[]):
         if currency != self.currency:
             raise CurrencyMismatch
 
-        # All relevant data is in data['data']
-        data = data['data']
-        start_time = self._parse_dt(data['DataStartdate'])
-        end_time = self._parse_dt(data['DataEnddate'])
-        updated = self._parse_dt(data['DateUpdated'])
-
-        area_data = {}
-        # Loop through response rows
-        for r in data['Rows']:
-            row_start_time = self._parse_dt(r['StartTime'])
-            row_end_time = self._parse_dt(r['EndTime'])
-
-            # Loop through columns
-            for c in r['Columns']:
-                name = c['Name']
-                # If areas is defined and name isn't in areas, skip column
-                if areas and name not in areas:
-                    continue
-
-                # If name isn't in area_data, initialize dictionary
-                if name not in area_data:
-                    area_data[name] = {
-                        'values': [],
-                    }
-
-                # Time based and average, max, min etc rows are separated
-                # with 'IsExtraRow' -marking
-                if r['IsExtraRow']:
-                    # Update extra data to dictionary
-                    area_data[name][r['Name']] = self._conv_to_float(c['Value'])
-                else:
-                    # Append dictionary to value list
-                    area_data[name]['values'].append({
-                        'start': row_start_time,
-                        'end': row_end_time,
-                        'value': self._conv_to_float(c['Value']),
-                    })
-
         return {
-            'start': start_time,
-            'end': end_time,
-            'updated': updated,
-            'currency': currency,
-            'areas': area_data
+            "start": start_time,
+            "end": end_time,
+            "updated": updated,
+            "currency": currency,
+            "areas": area_prices,
         }
 
-    def _fetch_json(self, data_type, end_date=None):
-        ''' Fetch JSON from API '''
+    def _get_url_params_areas(self, data_type, end_date=None, areas=None):
         # If end_date isn't set, default to tomorrow
         if end_date is None:
             end_date = date.today() + timedelta(days=1)
         # If end_date isn't a date or datetime object, try to parse a string
         if not isinstance(end_date, date) and not isinstance(end_date, datetime):
             end_date = parse_dt(end_date)
+        if areas is None:
+            areas = self.AREAS
+
+        endpoint = "DayAheadPrices"  # default to hourly
+        if data_type in [self.DAILY, self.WEEKLY, self.MONTHLY]:
+            endpoint = "AggregatePrices"
+        if data_type == self.YEARLY:
+            endpoint = "AggregatePrices/GetAnnuals"
+
+        api_url = f"{self.API_URL}/{endpoint}"
+        params = {
+            "currency": "EUR",
```

Comment:
Is this really intentional? Shouldn't prices should be fetch in the selected currency?